### DISCUSSION TITLE
Lesson 4 Challenge, can't get right answer.

### DISCUSSION THREAD

**Discussion by SalikJamal** [Cyfrin/foundry-full-course-cu#380] [Q&A]
I'm doing the Lesson 4 challenge, I was trying to learn how the numbers are working so was testing different numbers, checking different discussions, doing calculations etc. Somehow I got the right answer and minted the NFT on my main wallet. 

But then I wanted to replicate it just to be sure if I understood the challenge because that's what matters. I then switched on to development wallet and started calculations again. 

So currently as the writing of this post the value of ETH / USD from chainlink datafeed is: `190890850250`.
The operations in the function then 

```solidity
function solveChallenge(uint256 priceGuess, string memory yourTwitterHandle) external {
        uint256 actualPrice = getPrice(); // 190890850250
        
        // getDecimals() is always 8 checked in the contract.
        if (getDecimals() == 8) {
            actualPrice = actualPrice * 10e10; // The price at this point = 19089085025000000000000
        }
        if (actualPrice + 3e18 > priceGuess && actualPrice - 3e18 < priceGuess) {
            _updateAndRewardSolver(yourTwitterHandle);
        } else {
            revert LessonFour__WrongPrice();
        }
    }
```
    
Now according to the condition the answer 19089085025000000000000 should be working and satisfying the condition as it is less than `actualPrice + 3e18` and greater than `actualPrice - 3e18`. But still the transaction gives me "unable to estimate gas" error. 

Can someone tell me what am I doing wrong? I replicated the contract and values in local remix and it satisfies the condition. But somehow in the challenge contract I can't solve it. I also used chatGPT which says the answer I gave is correct as well.

**Comment by gulshanpr** [comment]
hey @SalikJamal , this might help you #239
